src/main/kotlin/com/openlattice/indexing/BackgroundLinkingIndexingService.kt

```diff
@@ -103,54 +105,117 @@ class BackgroundLinkingIndexingService(
-                    } catch (ex: Exception) {
-                        logger.error("Unable to index linking entity with from batch of linking ids $linkingIds.", ex)
-                    } finally {
-                        unLock(linkingIds)
-                    }
+    @Scheduled(fixedRate = LINKING_INDEX_RATE)
+    private fun linkingIndexing() {
+        if (isLinkingIndexingEnabled()) {
+            executor.submit {
+                try {
+                    logger.info("Starting background linking indexing task.")
+                    val esw = Stopwatch.createStarted()
+                    generateSequence(indexCandidates::poll)
+                            .chunked(LINKING_INDEX_SIZE)
+                            .forEach { candidateBatch ->
+                                limiter.acquire()
+                                val (linkingEntityKeyIdsWithLastWrite, linkingIds) = mapCandidates(candidateBatch)
+
+                                try {
+                                    index(linkingEntityKeyIdsWithLastWrite, linkingIds)
+                                } catch (ex: Exception) {
+                                    logger.error("Unable to index batch of linking ids $linkingIds.", ex)
+                                } finally {
+                                    unLock(linkingIds)
+                                    limiter.release()
+                                }
+                            }
+                    logger.info(
+                            "Finished background linking indexing task in ${esw.elapsed(TimeUnit.MILLISECONDS)} ms."
+                    )
+                } catch (ex: DistributedObjectDestroyedException) {
+                    logger.error("Linking indexing queue destroyed.", ex)
                 }
+            }
+        }
     }
 
     @Suppress("UNUSED")
-    private val linkingUnIndexingWorker = executor.submit {
-        generateSequence(unIndexCandidates::take)
-                .chunked(LINKING_INDEX_SIZE)
-                .asStream()
-                .parallel()
-                .forEach { candidateBatch ->
-                    if (!indexerConfiguration.backgroundLinkingIndexingEnabled) {
-                        return@forEach
-                    }
-
-                    val (linkingEntityKeyIdsWithLastWrite, linkingIds) = mapCandidates(candidateBatch)
-
-                    try {
-                        lock(linkingIds)
-                        unIndex(linkingEntityKeyIdsWithLastWrite, linkingIds)
-                    } catch (ex: Exception) {
-                        logger.error(
-                                "Unable to un-index linking entity with from batch of linking ids $linkingIds.", ex
-                        )
-                    } finally {
-                        unLock(linkingIds)
-                    }
+    @Scheduled(fixedRate = LINKING_INDEX_RATE)
+    private fun linkingUnIndexing() {
+        if (isLinkingIndexingEnabled()) {
```

Comment:
Same comment as above

Reply:
(and same comment below)

Reply:
This also looks nearly identical to the function above -- it seems preferable to combine these into one